documentation/docs/20-core-concepts/50-state-management.md

```diff
@@ -0,0 +1,46 @@
+---
+title: State Management
+---
+
+Managing state is one of the hardest parts in application development. This section covers various use cases with regards to state management and what to watch out for.
+
+## Avoid global state in SSR
+
+If you are creating an [SPA](glossary#csr) with SvelteKit, you can create global state freely, as you can be sure that it's only initialized inside the user's browser. If you use [SSR](glossary#ssr) however, you have to watch out for a couple of things when managing state. In many server environments, a single instance of your app will serve multiple users (this is not specific to SvelteKit - it's one of the gotchas of working with such environments). For that reason, per-request or per-user state must not be stored in global variables.
+
+Consider the following example where the user is set from inside a `load` function:
+
+```js 
+/// file: +page.js
+// @filename: ambient.d.ts
+declare module '$lib/user' {
+	export const user: { set: (value: any) => void };
+}
+
+// @filename: index.js
+// ---cut---
+// DON'T DO THIS!
+import { user } from '$lib/user';
+
+/** @type {import('./$types').PageLoad} */
+export async function load({ fetch }) {
+	const response = await fetch('/api/user');
+	user.set(await response.json());
+}
+```
+
+If you are using SSR, the `load` function will run on the server initially, which means that the whole server instance which serves _all_ requests from _all_ users has its `user` state set to the one just requested from the API. To scope this to a single user, you have a couple of options:
+
+- if you need to access the user state only inside server `load` functions, use [`locals`](hooks#server-hooks-handle)
+- if you need to persist the user state across reloads, but only need to access it inside `load` functions, use [`cookies` in server `load` functions](load#cookies-and-headers)
+- if you need to access and update the state inside components, use Svelte's [context feature](https://svelte.dev/docs#run-time-svelte-setcontext). That way, the state is scoped to components, which means they are not shared across different requests on the server. The drawback is that you can only set and subscribe to the store at component initialization. SvelteKit's stores from `$app/stores` for example are setup like this (which is why you may have encountered a related error message)
```

Comment:
users may not know what "component initialization" means. maybe we can clarify where it can be used (e.g. in a `<script>` tag, but not reactive statement - I don't know if that's exactly right but just to give an idea)